snackbar/snackbar/src/main/java/com/microsoft/device/dualscreen/snackbar/SnackbarContainer.kt

```diff
@@ -0,0 +1,174 @@
+/*
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License.
+ */
+
+package com.microsoft.device.dualscreen.snackbar
+
+import android.app.Activity
+import android.content.Context
+import android.util.AttributeSet
+import android.view.Gravity
+import android.view.ViewGroup.LayoutParams.MATCH_PARENT
+import android.view.ViewGroup.LayoutParams.WRAP_CONTENT
+import android.widget.FrameLayout
+import androidx.appcompat.app.AppCompatActivity
+import androidx.coordinatorlayout.widget.CoordinatorLayout
+import androidx.core.view.updateLayoutParams
+import androidx.window.layout.WindowInfoTracker
+import androidx.window.layout.WindowLayoutInfo
+import com.google.android.material.snackbar.Snackbar
+import com.microsoft.device.dualscreen.snackbar.SnackbarPosition.BOTH
+import com.microsoft.device.dualscreen.snackbar.SnackbarPosition.END
+import com.microsoft.device.dualscreen.snackbar.SnackbarPosition.START
+import com.microsoft.device.dualscreen.utils.wm.extractFoldingFeatureRect
+import com.microsoft.device.dualscreen.utils.wm.getWindowRect
+import com.microsoft.device.dualscreen.utils.wm.isFoldingFeatureVertical
+import com.microsoft.device.dualscreen.utils.wm.isInDualMode
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+/**
+ * Container used to display a [Snackbar] on the desired display area.
+ *
+ * How to use it:
+ *         Snackbar.make(snackbarContainer.coordinatorLayout, message, LENGTH_LONG)
+ *           .show(snackbarContainer, position)
+ *
+ */
+class SnackbarContainer @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null
+) : FrameLayout(context, attrs) {
+
+    companion object {
+        private const val COORDINATOR_LAYOUT_MARGIN = 20
+    }
+
+    val coordinatorLayout: CoordinatorLayout by lazy { CoordinatorLayout(context) }
+    private var job: Job? = null
+    private var windowLayoutInfo: WindowLayoutInfo? = null
+
+    init {
+        addView(coordinatorLayout, LayoutParams(MATCH_PARENT, WRAP_CONTENT))
+        registerWindowInfoFlow()
+    }
+
+    private fun requireActivity(): Activity = (context as? Activity)
+        ?: throw RuntimeException("Context must implement android.app.Activity")
+
+    private fun registerWindowInfoFlow() {
+        job = MainScope().launch {
```

Comment:
Should we also update this MainScope job to a lifecycle scope job to prevent receiving incorrect WindowLayoutInfo values?

Reply:
In this case is not really necessary, but I changed it to be in sync with the other components.